### PR TITLE
HMAN 228 HMAN 229 new database table hearing_channels

### PR DESCRIPTION
**Pull request by marvincudjoe**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/HMAN-228
https://tools.hmcts.net/jira/browse/HMAN-229

### Change description ###
New database table hearing_channels
Endpoints updated to allow new element channelType
- POST /hearing and PUT /hearing 
- GET /hearing and /hearings

**Does this PR introduce a breaking change?** (check one with "x")

```
[ x] Yes - hearingChannels is required in the payload
[ ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/hmc/helper/GetHearingResponseMapperTest.java [hunk 1]
```diff
@@ -286,6 +287,13 @@ void toHearingsResponseWhenListingStatusIsMissing() {
...
 
     @Test
+    void toHearingsResponseWhenDataIsPresentWithHearingChannel() {
+        HearingEntity hearingEntity = TestingUtil.getCaseHearingsEntity(PartyType.IND);
+        hearingEntity.getCaseHearingRequests().get(0).setHearingChannels(TestingUtil.hearingChannelsEntity());
+        GetHearingResponse response = getHearingResponseMapper.toHearingResponse(hearingEntity);
+        assertHearingChannel(response.getHearingDetails().getHearingChannels());
+    }
+
     void toHearingsResponseWhenRequestIdIsNull() {
```
Comment (markdathornehmcts) [inline src/test/java/uk/gov/hmcts/reform/hmc/helper/GetHearingResponseMapperTest.java:297]: Where is the @Test for this method now?

Comment (markdathornehmcts) [inline src/test/java/uk/gov/hmcts/reform/hmc/helper/GetHearingResponseMapperTest.java:297]: Re-add "@Test" to this test method

**Review by markdathornehmcts — CHANGES_REQUESTED** [review]
All good to me barring the 1 missing @Test annotation. Add that and if the pipeline builds successfully then approval will follow

File: src/main/java/uk/gov/hmcts/reform/hmc/model/hmi/Listing.java [hunk 2]
```diff
@@ -43,7 +44,7 @@ public class Listing {
 
     private List<ListingJoh> listingJohs;
 
-    private List<String> listingHearingChannels;
+    private List<HearingChannel> listingHearingChannels;
```
Comment (RebeccaBaker) [inline src/main/java/uk/gov/hmcts/reform/hmc/model/hmi/Listing.java:47]: same comment as above

File: src/main/java/uk/gov/hmcts/reform/hmc/model/HearingDetails.java [hunk 3]
```diff
@@ -76,6 +76,10 @@ public class HearingDetails {
     @Size(max = 70, message = ValidationError.AMEND_REASON_CODE_MAX_LENGTH)
     private String amendReasonCode;
 
+    @Valid
+    @NotNull(message = ValidationError.HEARING_CHANNEL_EMPTY)
+    private List<HearingChannel> hearingChannels;
```
Comment (RebeccaBaker) [inline src/main/java/uk/gov/hmcts/reform/hmc/model/HearingDetails.java:81]: same comment as above

File: src/main/java/uk/gov/hmcts/reform/hmc/model/CaseHearing.java [hunk 4]
```diff
@@ -52,4 +52,8 @@ public class CaseHearing {
 
     private Boolean hearingIsLinkedFlag;
 
+    @Size(max = 70)
+    @NotNull
+    private List<String> hearingChannels;
```
Comment (RebeccaBaker) [inline src/main/java/uk/gov/hmcts/reform/hmc/model/CaseHearing.java:57]: Is this correct, only allow 70 items in the list?

**Review by RebeccaBaker — APPROVED** [review]
Approved with a minor comment

**Review by markdathornehmcts — APPROVED** [review]

**Pull request merged**